.github/workflows/esrupgrade.yml

```diff
@@ -0,0 +1,92 @@
+name: ESR Upgrade
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+      - cloud
+      - release-*
+jobs:
+  test-esr-upgrade:
+    runs-on: ubuntu-latest-8-cores
+    env:
+      COMPOSE_PROJECT_NAME: ghactions
+      BUILD_IMAGE: mattermost/mattermost-enterprise-edition:7.8
+      LATEST_MIGRATION: 101
+      DB_DUMP_URL: https://lt-public-data.s3.amazonaws.com/47K_537_mysql_collationfixed.sql.gz
+      MYSQL_CONN_ARGS: -h localhost -P 3306 --protocol=tcp -ummuser -pmostest mattermost_test
+    steps:
+      - name: Checkout mattermost-server
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+      - name: Run docker compose
+        run: |
+          cd build
+          docker-compose --no-ansi run --rm start_dependencies
+          cat ../tests/test-data.ldif | docker-compose --no-ansi exec -T openldap bash -c 'ldapadd -x -D "cn=admin,dc=mm,dc=test,dc=com" -w mostest';
+          docker-compose --no-ansi exec -T minio sh -c 'mkdir -p /data/mattermost-test';
+          docker-compose --no-ansi ps
+      - name: Wait for docker compose
+        run: |
+          until docker network inspect ghactions_mm-test; do echo "Waiting for Docker Compose Network..."; sleep 1; done;
```

Comment:
You can use Github Actions  [containerised services](https://docs.github.com/en/actions/using-containerized-services/about-service-containers) in order to test this . 

You can check example [here](https://github.com/mattermost/mattermost-plugin-playbooks/blob/master/.github/workflows/ci.yaml#L292) 
I do not know if this suits your case but it is easier to maintain and you can spawn as many as you want without worrying if services are up etc

Reply:
That's interesting, I'll take a look and see if it can be applied. In any case, I'd defer that to another PR, since this bit was copied from another action and it's been proved to work.